torch/utils/data/datapipes/utils/common.py

```
@@ -27,6 +34,31 @@ def check_lambda_fn(fn):
         )
 
 
+def serialize_fn(fn, is_dill_available):
+    """
+    Try to serialize ``fn`` using `pickle`, falls back to `dill` if `pickle` fails and DILL_AVAILABLE.
+    Returns a tuple of serialized function and SerializationType indicating the serialization method.
+    """
+    try:
+        return pickle.dumps(fn), SerializationType("pickle")
+    except (pickle.PickleError, AttributeError):
```

I wonder if I should catch all Exception here.